### PR TITLE
refactor(q8): resolve FFN decode chain route

### PR DESCRIPTION
**Pull request by timtoole02**
## Summary
- split x86 Q8 FFN decode-chain route selection into a dedicated resolver that returns the packed gate/up/down handles and validated widths before execution starts
- keep the existing VNNI/rawptr down-projection execution path, but drive it from the resolved route object instead of re-checking the tensor graph inline
- add a fail-closed unit test for incomplete runtime-packed FFN decode-chain inputs and keep the existing VNNI-backed decode-chain coverage intact

## Verification
- `cargo fmt --check`
- `cargo test q8_ffn_down_consumer_matches_runtime_packed_baseline -- --nocapture`

## Notes
- Fresh Ubuntu Linux x86_64 same-host rerun on current base `07c912cf3041` vs PR head `40d22548ec9b` preserved the marker guard and the focused FFN-down unit coverage.
- The rerun did not show a meaningful speedup: Camelid TTFT `2423.11 ms -> 2429.53 ms` and total `2423.45 ms -> 2429.75 ms` (`+0.26%` on both), so this PR should stay framed as a refactor/route-shape slice only, not a perf-improvement claim.

### CODE REVIEW

**Comment by timtoole02** [comment]
Ubuntu x86_64 validation refresh for `40d22548ec9bfbe3361dc344a9e74178daa4c8a6`:

Validation:
- `cargo fmt --all -- --check`
- `cargo test --lib x86_q8_ffn_decode_chain_route_resolver_fails_closed_on_incomplete_inputs -- --nocapture`
- `cargo test --lib q8_ffn_decode_chain_uses_vnni_down_when_gated -- --nocapture`
- `cargo test --lib q8_ffn_down_consumer_matches_runtime_packed_baseline -- --nocapture`
- `cargo build --release`

Fresh Ubuntu same-host evidence:
- One-token parity against llama.cpp passed for prompt `Reply with exactly one capital letter: C`.
- Prompt tokens matched.
- Generated token `[66]` matched.
- Generated text `c` matched.

Bounded same-host 4-token timing under the current default-off FFN decode-chain gate set:
- Camelid TTFT: `341.26 ms`
- llama.cpp TTFT: `228.56 ms`
- Camelid total elapsed: `341.47 ms`
- llama.cpp total elapsed: `355.53 ms`
- Camelid backend first content: `129 ms`
- Camelid backend generate: `273 ms`

Route proof from the measured run:
- `ffn_decode_chain_taken=84`
- `ffn_decode_chain_total_us=98134`
- `ffn_decode_chain_down_us=34586`
- `ffn_gate_up.decode_fused_activation.calls=84`
- `ffn_down.x86_decode_consumer.calls=84`
- `logits.x86_output_decode_owner.calls=3`

Guardrail note:
- The benchmark captured non-empty streamed output from both engines, but the expected `CMLD-BENCH` marker was not present in either measured run. Retain this as bounded route/timing smoke for the refactor, not as promotion evidence.

**Comment by timtoole02** [comment]
Ubuntu x86_64 current-main comparison refresh for `40d22548ec9bfbe3361dc344a9e74178daa4c8a6` against current `main` `1685c65aeddac1317a0b477503bfc31bd0a1dd0b`.

Validation on the candidate slice passed:
- `cargo fmt --all -- --check`
- `cargo test --lib x86_q8_ffn_decode_chain_route_resolver_fails_closed_on_incomplete_inputs -- --nocapture`
- `cargo test --lib q8_ffn_decode_chain_uses_vnni_down_when_gated -- --nocapture`
- `cargo build --release --bin camelid`
- `cargo build --release --bin camelid` on current `main`

Fresh bounded same-host 8-token timing, identical harness settings, guardrails passed on both runs:
- Current `main`: Camelid TTFT `2400.36 ms`, total elapsed `2400.65 ms`; llama.cpp TTFT `213.02 ms`, total elapsed `402.37 ms`
- PR #123 candidate: Camelid TTFT `2427.16 ms`, total elapsed `2427.43 ms`; llama.cpp TTFT `214.21 ms`, total elapsed `403.39 ms`
- Candidate minus current `main`: Camelid `+26.80 ms` TTFT, `+26.78 ms` total; llama.cpp drift in the paired control run was only `+1.19 ms` TTFT and `+1.02 ms` total

Decision:
- Retain this PR as a code-hygiene/refactor slice only.
- Reject it as a retained performance win or timing-improvement slice: the fresh same-host comparison did not beat current `main` and instead measured a small slowdown.
- No support/default-on/throughput promotion is justified from this run.

**Comment by timtoole02** [comment]
Follow-up on the failing `public-scrub` status for this PR:

- I pushed `07c912c` to `main` to remove the remaining raw SSH placeholder patterns from the inherited public evidence bundle command files.
- That repair passed `bash scripts/check-public-scrub.sh` on current `main` before push.

Next action for this branch:
- rebase/refresh onto current `main`
- rerun CI on the updated merge base

The fresh Ubuntu same-host result from this lane does not change the performance conclusion: keep PR #123 as a refactor/code-hygiene slice, not as a retained speedup claim.

**Comment by timtoole02** [comment]
Fresh Linux x86_64 same-host evidence for this refactor landed in artifact `cron-95495a91-20260524T1040Z-pr123-samehost`.

Measured on the PR head versus its base commit:
- Camelid TTFT: base `7815.95 ms`, head `7845.56 ms` (`+29.61 ms`, about `+0.38%`)
- Camelid total elapsed: base `7816.18 ms`, head `7845.77 ms` (`+29.59 ms`, about `+0.38%`)
- llama.cpp reference stayed effectively flat: base `391.74 / 584.63 ms`, head `389.47 / 582.54 ms` for TTFT / total
- One-token parity on the PR head matched exactly: prompt tokens matched, generated tokens matched, backend text `"Hello"`, llama.cpp text `"Hello"`

Interpretation: this looks performance-neutral within run noise for the active Llama 3.2 3B Q8_0 Ubuntu x86 lane, so the refactor does not show a measurable same-host regression from this artifact.

Validation note for this lane: because Ubuntu runs now use a shared `CARGO_TARGET_DIR` under disk-guard policy, the same-host harness had to launch the backend from the shared-target binary path rather than a per-worktree `target/release` path.

**Comment by timtoole02** [comment]
Fresh Ubuntu Linux x86_64 same-host rerun on current base `07c912cf3041` vs PR head `40d22548ec9b`:

- Head unit check passed: `cargo test q8_ffn_down_consumer_matches_runtime_packed_baseline --lib`
- Base same-host benchmark: Camelid TTFT `2423.11 ms`, total `2423.45 ms`; llama.cpp TTFT `232.38 ms`, total `423.22 ms`
- Head same-host benchmark: Camelid TTFT `2429.53 ms`, total `2429.75 ms`; llama.cpp TTFT `243.64 ms`, total `434.49 ms`
- Marker guard passed on both measured runs
- PR-head delta vs base: `+6.42 ms` TTFT (`+0.26%`), `+6.30 ms` total (`+0.26%`)

Decision: this rerun does not show a meaningful speedup over current `main`, so I would keep PR #123 framed as a refactor/route-shape slice only and not attach a perf-improvement claim to it from this evidence.

**Comment by timtoole02** [comment]
Closing as superseded by #140, which keeps the FFN decode-chain planner wiring as the current review target.
